smartsim/_core/_install/platform.py

```diff
@@ -0,0 +1,241 @@
+        """Find the enum based on environment CUDA
+
+        :return: Enum for the version of CUDA currently available
+        """
+        if cuda_home := os.environ.get("CUDA_HOME"):
+            cuda_path = pathlib.Path(cuda_home)
+            with open(cuda_path / "version.json", "r", encoding="utf-8") as file_handle:
+                cuda_versions = json.load(file_handle)
+            major, minor = cuda_versions["cuda"]["version"].split(".")[0:2]
+            return cls.from_str(f"cuda-{major}.{minor}")
+        return None
+
+    @classmethod
+    def detect_rocm_version(cls) -> t.Optional["Device"]:
+        """Find the enum based on environment ROCm
+
+        :return: Enum for the version of ROCm currently available
+        """
+        if rocm_home := os.environ.get("ROCM_HOME"):
+            rocm_path = pathlib.Path(rocm_home)
+            fname = rocm_path / ".info" / "version"
+            with open(fname, "r", encoding="utf-8") as file_handle:
+                major, minor = file_handle.readline().split("-")[0].split(".")
+            return cls.from_str(f"rocm-{major}.{minor}")
+        return None
+
+    def is_gpu(self) -> bool:
+        """Whether the enum is categorized as a GPU
+
+        :return: True if GPU
+        """
+        return self != type(self).CPU
+
+    def is_cuda(self) -> bool:
+        """Whether the enum is associated with a CUDA device
+
+        :return: True for any supported CUDA enums
+        """
+        cls = type(self)
+        return self in cls.cuda_enums()
+
+    def is_rocm(self) -> bool:
+        """Whether the enum is associated with a ROCm device
+
+        :return: True for any supported ROCm enums
+        """
+        cls = type(self)
+        return self in cls.rocm_enums()
+
+    @classmethod
+    def cuda_enums(cls) -> t.Tuple["Device", ...]:
+        """Detect all CUDA devices supported by SmartSim
+
+        :return: all enums associated with CUDA
+        """
+        return tuple(device for device in cls if "cuda" in device.value)
+
+    @classmethod
+    def rocm_enums(cls) -> t.Tuple["Device", ...]:
+        """Detect all ROCm devices supported by SmartSim
+
+        :return: all enums associated with ROCm
+        """
+        return tuple(device for device in cls if "rocm" in device.value)
+
+
+class OperatingSystem(enum.Enum):
+    """Enum for all supported operating systems
+
+    :raises UnsupportedError: If operating system is not not supported
+    """
+
+    LINUX = "linux"
+    DARWIN = "darwin"
+
+    @classmethod
+    def from_str(cls, string: str, /) -> "OperatingSystem":
+        """Return enum associated with the OS
+
+        :param string: String representing the OS
+        :raises UnsupportedError: If not enumerated here, unsupported
+        :return: Enum for a specific OS
+        """
+        string = string.lower()
+        for type_ in cls:
+            if string in type_.value:
+                return type_
```

Comment:
I think we can just do a direct string check since they're all single valued Enums

Reply:
That's a good point! After you mentioned it, I bothered to do the first bit of research into how python handles linux, and it doesn't look like python has appended the kernel version for linux since like version 3.3, hahaha!

Direct equality checks should be plenty sufficient!